### PR TITLE
Annotations for routing configuration

### PR DESCRIPTION
**Pull request by aquibbaig**
This PR is regarding issue #296 . Please review and check it.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (agathver) [inline composer.json]: This file is unnecessary

File: site/src/Librecores/SiteBundle/Resources/config/routing.yml [hunk 2]
```diff
@@ -1,10 +1,7 @@
 librecores_site_home:
-    path:     /
-    defaults: { _controller: LibrecoresSiteBundle:Default:home }
+    resource: '@LibrecoresSiteBundle/Controller/'
```
Comment (agathver) [inline site/src/Librecores/SiteBundle/Resources/config/routing.yml]: you can actually write:

```yaml
librecores_site:
    resource: '@LibrecoresSiteBundle/Controller/'
    type: annotation
```

_Once_. Repetition is redundant

File: site/src/Librecores/SiteBundle/Controller/DefaultController.php [hunk 3]
```diff
@@ -40,8 +43,10 @@ public function homeAction(Request $request)
     /**
      * XXX: add caching for static pages
      * see http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/cache.html
+     *
+     * @Route("/{page}", name="librecores_site_page", requirements={"page"=".+"})
```
Comment (imphil) [inline site/src/Librecores/SiteBundle/Controller/DefaultController.php]: The comment ` # allow / inside path` was pretty helpful to understand why the `requirements` option is added. Please restore that. Can you also add spaces before/after the `=` to make it `requirements = {` etc.

Comment (aquibbaig) [reply site/src/Librecores/SiteBundle/Controller/DefaultController.php]: @imphil Thank you for suggesting! I have made the changes. Please review them.

**Comment by imphil** [comment]
Thanks @aquibbaig! I've squashed and merged your changes. Hooray on your first contribution to this repository!

**Pull request merged**